Automattic/pocket-casts-android
# Bookmarks - Add to episode details page

## Description

This adds bookmarks on the episode details page.

** Note that some of the components like header row,  no bookmarks view, plus upsell view and loading view have been extracted outside `BookmarksFragment` to reduce the class size and make it more readable. This + adding a few components to `showkase` (as requested in https://github.com/Automattic/pocket-casts-android/pull/1210#discussion_r1275927045) has increased PR size.

I also wanted to move `BookmarksFragment` to the bookmark package and rename file `BookmarkItem` to  `BookmarkRow` but for some reason doing that created new files instead of showing them as changes on GitHub. I will do that before merging this PR.

## Testing Instructions
1. Install debug build
2. Open a podcast episode with a few bookmarks added to it
3. ✅ Notice that bookmarks tab is shown 
4. Tap on the bookmarks tab
5. ✅ Notice that the bookmarks page is shown
6. Long press a bookmark row
7. ✅ Notice that the multi-select toolbar appears
8. Dismiss the dialog
9. ✅ Notice that the multi-select toolbar is not shown

#### Other tests

1. The screen reuses `MultiSelectEpisodeHelper`. Smoke test that the multi-select operations work correctly on the screen. (** edit action is not yet functional)

2. Disable the bookmarks feature and test that the bookmarks page cannot be accessed on the episode details screen (no tabs and no scroll to the page).

#### Notes

1. Bookmark sorting works but it reuses the preferences for the playing episode screen. Since they both apply to a single selected episode, maybe I should continue using it. I'll address it in a later PR.
2. Header row, bookmark row, no bookmarks view, and plus upsell view have been reused from the Player screen with color changes. Plus upsell view logo needs to be updated for lighter themes. I'll address it when I integrate Plus upsell screens.
3. Design - Known issues:
    - Episode Details screen layout changes (date and duration) not yet integrated 
    - Header row option needs to go next to the tabs 
    - Tabs design does not match Figma

## Screenshots or Screencast 

https://github.com/Automattic/pocket-casts-android/assets/1405144/7ae3962a-5308-497e-82f3-98fc61f7634f

## Checklist
 
#### I have tested any UI changes...
